### DISCUSSION TITLE
What am I doing wrong? Trying to load Wizard-vicuna-13B-GGML. I can't get any GGML to work!

### DISCUSSION THREAD

**Discussion by viperwasp** [oobabooga/text-generation-webui#2558] [General]
GPU - RTX 4080
RAM - 64GB

1. I downloaded and installed Oobabooga months ago. I keep it updated via "git pull https://github.com/oobabooga/text-generation-webui.git"

2. I downloaded the model from "https://huggingface.co/TheBloke/wizard-vicuna-13B-GGML"

3. I put it in a folder called "Wizard-vicuna-13B-GGML" located in text-generation-webui > models 
3B. There are no other files in that folder. 
4. I used the following arguments to launch this model. 
python server.py --auto-devices --chat --model Wizard-vicuna-13B-GGML --wbits 4 --groupsize 128 --model_type llama 

Here is my error 
Traceback (most recent call last):
  File "C:\AI\oobabooga-windowsBest\text-generation-webui\server.py", line 1079, in <module>
    shared.model, shared.tokenizer = load_model(shared.model_name)
  File "C:\AI\oobabooga-windowsBest\text-generation-webui\modules\models.py", line 94, in load_model
    output = load_func(model_name)
  File "C:\AI\oobabooga-windowsBest\text-generation-webui\modules\models.py", line 294, in AutoGPTQ_loader
    import modules.AutoGPTQ_loader
  File "C:\AI\oobabooga-windowsBest\text-generation-webui\modules\AutoGPTQ_loader.py", line 3, in <module>
    from auto_gptq import AutoGPTQForCausalLM, BaseQuantizeConfig
ModuleNotFoundError: No module named 'auto_gptq'

5. I added the following argument --gptq-for-llama 

So now 
" python server.py --auto-devices --chat --model Wizard-vicuna-13B-GGML --wbits 4 --groupsize 128 --model_type llama  --gptq-for-llama"

And I get this error 
Starting the web UI...
INFO:Gradio HTTP request redirected to localhost :)
INFO:Loading Wizard-vicuna-13B-GGML...
ERROR:Could not find the quantized model in .pt or .safetensors format, exiting...
Press any key to continue . . .

6. I don't know what to do? I don't know how to setup oobabooga for GGML. I have never got GGML to work? 
I tried going here. 
https://github.com/oobabooga/text-generation-webui/blob/main/docs/llama.cpp-models.md

I went into Oobabooga directory and did CMD. 
pip uninstall -y llama-cpp-python
set CMAKE_ARGS="-DLLAMA_CUBLAS=on"
set FORCE_CMAKE=1
pip install llama-cpp-python --no-cache-dir

As stated in the guide. It did stuff? But I don't know what to do now. The errors I listed above are the error I'm getting. What am I missing?

**Comment by viperwasp** [comment]
The "fix" seemed to be to load the model without any arguments? Using the default loader. Make sure there is no groupsize or Wbits. I think it's working now.

**Comment by jparmstr** [comment]
Yeah, the command line arguments for GGML have seemed broken for a while, in my experience. The "wbits" and "groupsize" arguments are for GPTQ not GGML, so I think that was your first problem.

But I've also been experiencing the second problem, that GGML models can only be loaded via the UI, not command line arguments. If I try:
`python server.py --model llama-13b.ggml.q4_0 --model_type=llama --chat --sdp-attention`

I get:
`INFO:Loading llama-13b.ggml.q4_0...`
`ERROR:The path to the model does not exist. Exiting.`

**Reply by jparmstr** [reply]
Also, the procedure you did at the end for rebuilding Llama.cpp with CUBLAS support lets you offload some layers to your GPU. Llama.cpp was originally meant to run Llama models on the CPU, but now supports GPU offloading. In the UI, in the llama.cpp section under models, you can increase n-gpu-layers. There are 32 layers in Llama models. Based on your GPU you can probably fully offload that 13B model to the GPU and it should be pretty fast.

Remember to click "Reload the model" after making changes. If successful, you should get something like this in the command prompt:
`llama_model_load_internal: [cublas] offloading 32 layers to GPU`
`llama_model_load_internal: [cublas] total VRAM used: 3475 MB`

**Reply by viperwasp** [reply]
Thanks I can't get the GPU off loading to work. I tried it. That's one of my goals with GGML. In fact I mainly want to use 30B this way. I was already pleased with 13B on GPU only. However 30B is very slow splitting between CPU and GPU the old way. (4bit none GGML I mean. It only used like 1-2 threads.) It runs faster on CPU only. But as a few other people have experienced I can't get it to offload onto GPU at all. Going to slowly work on it but it will likely be ironed out or make easier later. Some troubleshooting steps I might be able to do. 

But thanks for the help jparmstr. I am going to read this more when I have time to make sure we are both on the same page. 30B is decent for me on CPU only. However I would love to make it a little faster with GPU offloading.
